### PR TITLE
#163388258 Users can follow and unfollow profiles

### PR DESCRIPTION
**Pull request by calebrotich**
#### What does this PR do?
This PR enables users to follow, unfollow and view their followers and following lists

#### Description of Task to be completed?
- Create tests
- Implement the `follow` logic
- Implement the `unfollow` logic
- List a user's followers

#### How should this be manually tested?
Make the following requests to postman:
*N/B:* {{BASE_URL}} = `http://127.0.0.1:8000` for localhost and `https://ah-bird-box-staging.herokuapp.com` for heroku

| Request         | URL                | Description |
| ------------- | ------------- |------------------|
| `POST`  | `{{BASE_URL}}/api/profile/<username>/follow/ ` |Follow a user with the specified username|
| `DELETE`  | `{{BASE_URL}}/profile/<username>/follow/`  |Unfollow a user with the specified username |
| `GET`  | `{{BASE_URL}}/api/profile/<username>/following/ ` |List followers and the following list of the user specified by the username|

#### What are the relevant pivotal tracker stories?
[#163388258](https://www.pivotaltracker.com/story/show/163388258)

#### Screenshots (if appropriate)
_*Follow a user*_
image

_*Follow a user you already follow*_
image

_*Unfollow a user*_
image

_*Unfollow a user you do not follow*_
image

_*List followers and unfollowers*_
image

### CODE REVIEW

File: authors/apps/profiles/views.py [hunk 1]
```diff
@@ -69,3 +75,120 @@ def put(self, request, username):
...
+                'message': str(e)
+            },
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        request.user.following.add(to_be_followed)
+        request.user.save()
+
+        serializer = FollowUnfollowSerializer(request.user)
+        message = {
+            "message": "Profile successfully followed",
+            "user": serializer.data
```
Comment (sam-karis) [inline authors/apps/profiles/views.py:126]: Make this message descriptive e.g. `You are now following John Doe`

File: authors/apps/profiles/views.py [hunk 2]
```diff
@@ -69,3 +75,120 @@ def put(self, request, username):
...
+                {
+                    'error': 'You do not follow this user'
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        follower.following.remove(id)
+        serializer = FollowUnfollowSerializer(follower)
+        message = {
+            "message": "Profile successfully unfollowed",
```
Comment (sam-karis) [inline authors/apps/profiles/views.py]: As above make the response message descriptive e.g.`You stopped following John Doe`

**Review by sam-karis — CHANGES_REQUESTED** [review]
Also, ensure you do not have unnecessary migration files e.g this two`authors/apps/profiles/migrations/0003_remove_profile_following.py` and `authors/apps/profiles/migrations/0002_profile_following.py` to one migration file

**Review by TooColline — COMMENTED** [review]
LGTM

**Review by bryan-munene — APPROVED** [review]
This good work. Ensure it works on Heroku

**Review by sam-karis — APPROVED** [review]
LGTM

**Pull request merged**